core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,288 @@
+const previewRestartButton = '.e2e-test-preview-restart-button';
+const stateConversationContent = '.e2e-test-conversation-content';
+const explorationCompletionToastMessage = '.e2e-test-lesson-completion-message';
+
+export class ExplorationEditor extends BaseUser {
+  /**
+   * Function to navigate to the creator dashboard page.
+   */
+  async navigateToCreatorDashboard(): Promise<void> {
+    await this.page.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Function to create an exploration in the Exploration Editor.
+   */
+  async createExploration(): Promise<void> {
+    await this.clickOn(createExplorationButton);
+    await this.clickOn(dismissWelcomeModalSelector);
+  }
+
+  /**
+   * Function to add content to a card.
+   * @param {string} questionText - The content to be added to the card.
+   */
+  async updateCardContent(questionText: string): Promise<void> {
+    await this.clickOn(stateEditSelector);
+    await this.waitForElementToBeClickable(stateContentInputField);
+    await this.page.click(stateContentInputField, {clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+    await this.type(stateContentInputField, `${questionText}`);
+    await this.clickOn(saveContentButton);
+    await this.page.waitForSelector(stateContentInputField, {hidden: true});
+  }
+
+  /**
+   * Function to add an interaction to the exploration.
+   * @param {string} interactionToAdd - The interaction type to add to the Exploration.
+   */
+  async addInteraction(interactionToAdd: string): Promise<void> {
+    await this.clickOn(addInteractionButton);
+    await this.clickOn(interactionToAdd);
+    await this.clickOn(saveInteractionButton);
+  }
+
+  /**
+   * Function to display the Oppia responses section.
+   */
+  async viewOppiaResponses(): Promise<void> {
+    await this.page.waitForSelector(stateResponsesSelector, {visible: true});
+    await this.page.click(stateResponsesSelector);
+  }
+
+  /**
+   * Function to select the card that learners will be directed to from the current card.
+   */
+  async oppiaDirectlearnersTo(card: string): Promise<void> {
+    await this.clickOn(openOutcomeDestButton);
+    await this.waitForElementToBeClickable(destinationCardSelector);
+    await this.page.select(destinationCardSelector, card);
+  }
+
+  /**
+   * Function to name a new card in the exploration.
+   */
+  async nameNewCard(cardName: string): Promise<void> {
+    await this.page.type(addStateInput, cardName);
+    await this.waitForElementToBeClickable(saveOutcomeDestButton);
+    await this.page.click(saveOutcomeDestButton);
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (!element) {
+        await this.clickOn(mobileOptionsButton);
+      }
+      await this.clickOn(mobileSaveChangesButton);
+      await this.clickOn(saveDraftButton);
```

Comment:
Pull common statements like this and the following line outside the if/else. In general try to reduce the branching.

Reply:
Done.